WordPress/src/main/java/org/wordpress/android/ui/suggestion/service/SuggestionService.java

```diff
@@ -103,24 +122,16 @@ private void removeSiteIdFromSuggestionRequestsAndStopServiceIfNecessary(long si
         }
     }
 
-    public void updateTags(final long siteId) {
-        if (mCurrentlyRequestingTagsSiteIds.contains(siteId)) {
-            return;
-        }
+    private void updateTags(final long siteId) {
         mCurrentlyRequestingTagsSiteIds.add(siteId);
-        RestRequest.Listener listener = new RestRequest.Listener() {
-            @Override
-            public void onResponse(JSONObject jsonObject) {
-                handleTagsUpdatedResponse(siteId, jsonObject);
-                removeSiteIdFromTagRequestsAndStopServiceIfNecessary(siteId);
-            }
+        RestRequest.Listener listener = jsonObject -> {
+            handleTagsUpdatedResponse(siteId, jsonObject);
+            TAG_REQUEST_THROTTLER.onResponseReceived(siteId);
```

Comment:
This line is the only change to the logic in the tags listeners. The rest of the diff is just the result of switching to using the lambda syntax.